src/test/java/uk/gov/hmcts/dts/fact/services/admin/AdminCourtGeneralInfoServiceTest.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.dts.fact.services.admin;
+
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import uk.gov.hmcts.dts.fact.config.security.RolesProvider;
+import uk.gov.hmcts.dts.fact.entity.Court;
+import uk.gov.hmcts.dts.fact.model.admin.CourtGeneralInfo;
+import uk.gov.hmcts.dts.fact.repositories.CourtRepository;
+
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.hmcts.dts.fact.services.admin.AdminRole.FACT_ADMIN;
+
+@ExtendWith({SpringExtension.class, MockitoExtension.class})
+@ContextConfiguration(classes = AdminCourtGeneralInfoService.class)
+public class AdminCourtGeneralInfoServiceTest {
+    private static final String COURT_SLUG = "some slug";
+    private static final String COURT_ALERT = "English alert";
+    private static final String COURT_ALERT_CY = "Welsh alert";
+    private static final CourtGeneralInfo COURT_GENERAL_INFO = new CourtGeneralInfo(false, null, null, null, COURT_ALERT, COURT_ALERT_CY);
```

Comment:
Is it normal for these variables to be set as null? Looks strange to me, but might be normal depending on how the entity stuff works

Reply:
If it is admin but not superadmin, these values are always null because admin can only set Alert messages but not the rest. I have got a test for superadmin...think it is in the endpoint test